app/services/xcode_manager_service.rb

```diff
@@ -0,0 +1,170 @@
+require_relative "../shared/logging_module"
+require_relative "./services"
+
+require "xcode/install"
+
+module FastlaneCI
+  # Manages Xcode installations
+  class XcodeManagerService
+    include FastlaneCI::Logging
+
+    # Keeps an a hash, the key being the Gem::Version of the Xcode version we're installing
+    # and the value being the % of the download progress
+    attr_accessor :installations_in_progress
+
+    def initialize
+      if ENV["XCODE_INSTALL_USER"].to_s.length == 0
+        raise "No `XCODE_INSTALL_USER` ENV variable provided, please provide one when launching up the fastlane.ci"
+      end
+
+      self.installations_in_progress = {}
+    end
+
+    # A shared reference to the `XcodeInstall::Installer` object we use
+    def installer
```

Comment:
If we're deleting the comment `# @_installer.rm_list_cache`, could we please use a memoized method?

```ruby
def installer
  @_installer ||= Installer.new
end
```

Reply:
Yeah, it's an open task to take a look at if we should call this method